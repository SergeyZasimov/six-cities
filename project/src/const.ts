enum AppRoute {
  Main = '/',
  Login = '/login',
  Favorites = '/favorites',
  Room = '/offer',
  NotFound = '*',
}

enum AuthorizationStatus {
  Auth = 'AUTH',
  NoAuth = 'NO_AUTH',
  Unknown = 'UNKNOWN',
}

enum IconUrl {
  Default = './img/pin.svg',
  Active = './img/pin-active.svg',
}

enum CardType {
  Cities = 'cities',
  Favorites = 'favorites',
  NearPlaces = 'near-places',
}

enum MapType {
  Cities = 'cities',
  Property = 'property',
}

enum ApiRoute {
  Offers = '/hotels',
  Login = '/login',
  Logout = '/logout',
  Comments = '/comments',
}

enum DomainNameSpace {
  User = 'User',
  Data = 'Data',
  Location = 'Location',
}

const SortType = {
  Popular: 'Popular',
  PriceLowToHigh: 'Price: low to high',
  PriceHighToLow: 'Price: high to low',
  TopRatedFirst: 'Top rated first',
} as const;

const StateAction = {
  Location: {
    ChangeLocation: 'location/changeLocation',
  },
  Data: {
    LoadOffers: 'data/loadOffers',
    LoadRoom: 'data/loadRoom',
    SendNewComment: 'data/sendNewComment',
  },
  User: {
    CheckAuth: 'user/checkAuth',
    Login: 'user/login',
    Logout: 'user/logout',
    RedirectToRoute: 'user/redirectToRoute',
  },
} as const;

const DEFAULT_CITIES = [
  'Paris',
  'Cologne',
  'Brussels',
  'Amsterdam',
  'Hamburg',
  'Dusseldorf',
] as const;

const DEFAULT_CITY_NAME = 'Paris';
const BACKEND_URL = 'https://10.react.pages.academy/six-cities';
const REQUEST_TIMEOUT = 5000;
const AUTH_TOKEN_KEY_NAME = 'six-cities-token';
const MAX_RATING = 5;
const MAX_COMMENT_LENGTH = 300;
const MIN_COMMENT_LENGTH = 50;


export {
  AppRoute,
  AuthorizationStatus,
  IconUrl,
  CardType,
  MapType,
  ApiRoute,
  DomainNameSpace,
  SortType,
  StateAction,
  DEFAULT_CITIES,
  DEFAULT_CITY_NAME,
  BACKEND_URL,
  REQUEST_TIMEOUT,
  AUTH_TOKEN_KEY_NAME,
  MAX_RATING,
  MAX_COMMENT_LENGTH,
  MIN_COMMENT_LENGTH,
};
