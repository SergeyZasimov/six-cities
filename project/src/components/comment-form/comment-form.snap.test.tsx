import { render } from '@testing-library/react';
import CommentForm from './comment-form';
import { Provider } from 'react-redux';
import { configureMockStore } from '@jedmao/redux-mock-store';
import { SendingStatus } from '../../const';


const mockStore = configureMockStore();


describe('CommentForm: Snapshot', () => {
  const store = mockStore({
    Comments: {
      status: SendingStatus.Idle
    }
  });

  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <CommentForm roomId={1} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
