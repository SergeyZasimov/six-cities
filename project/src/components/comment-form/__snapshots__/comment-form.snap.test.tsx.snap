// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm: Snapshot should render correctly 1`] = `
<div>
  <form
    action="#"
    class="reviews__form form"
    method="post"
  >
    <label
      class="reviews__label form__label"
      for="review"
    >
      Your review
    </label>
    <div
      class="reviews__rating-form form__rating"
    >
      <input
        class="form__rating-input visually-hidden"
        data-testid="rating-star-5"
        id="5-star"
        name="rating"
        type="radio"
        value="5"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="5-star"
        title="perfect"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        data-testid="rating-star-4"
        id="4-star"
        name="rating"
        type="radio"
        value="4"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="4-star"
        title="perfect"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        data-testid="rating-star-3"
        id="3-star"
        name="rating"
        type="radio"
        value="3"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="3-star"
        title="perfect"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        data-testid="rating-star-2"
        id="2-star"
        name="rating"
        type="radio"
        value="2"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="2-star"
        title="perfect"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
      </label>
      <input
        class="form__rating-input visually-hidden"
        data-testid="rating-star-1"
        id="1-star"
        name="rating"
        type="radio"
        value="1"
      />
      <label
        class="reviews__rating-label form__rating-label"
        for="1-star"
        title="perfect"
      >
        <svg
          class="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlink:href="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      class="reviews__textarea form__textarea"
      data-testid="comment-text"
      id="review"
      maxlength="300"
      name="review"
      placeholder="Tell how was your stay, what you like and what can be improved"
    />
    <div
      class="reviews__button-wrapper"
    >
      <p
        class="reviews__help"
      >
        To submit review please make sure to set 
        <span
          class="reviews__star"
        >
          rating
        </span>
         and describe your stay with at least 
        <b
          class="reviews__text-amount"
        >
          50
           characters
        </b>
        .
      </p>
      <button
        class="reviews__submit form__submit button"
        data-testid="comment-submit-button"
        disabled=""
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
